app/filters/common.py

```diff
@@ -29,6 +29,29 @@ class NameFilterMixin:
     name__ilike: str | None = None
 
 
+class ILikeSearchFilterMixin:
+    def __init_subclass__(cls, **kwargs):
+        """Add ilike search on multiple columns.
+
+        Ensure that this mixin is added to a filter corresponding to a model with
+        ILIKE_SEARCH_FIELDS available.
+        """
+        # Add filter key and annotation for ilike_search
+        cls.__annotations__[ILIKE_SEARCH_FIELD_NAME] = str | None
+        setattr(cls, ILIKE_SEARCH_FIELD_NAME, None)
+
+        # Set fastapi-filter custom serach name and fields
+        # See example: https://github.com/arthurio/fastapi-filter/blob/8c07dd55dfa63f09ae70eb980d51714323809906/examples/fastapi_filter_mongoengine.py#L91-L92
+        cls.Constants.search_field_name = ILIKE_SEARCH_FIELD_NAME  # pyright: ignore [reportAttributeAccessIssue]
+
+        # allow search_model_fields overrides
+        if hasattr(cls.Constants, "search_model_fields") and cls.Constants.search_model_fields:  # pyright: ignore [reportAttributeAccessIssue]
```

Comment:
Minor, it should be equivalent to:

```suggestion
        if getattr(cls.Constants, "search_model_fields", None):
```